pkg/command/root_test.go

```diff
@@ -908,3 +903,311 @@ func TestCompletionShortHelpInActiveHelp(t *testing.T) {
+					target:    configtypes.TargetK8s,
+					invokedAs: []string{"dummy"},
+					aliases:   []string{"dum"},
+				},
+			},
+			args:     []string{"dummy", "say", "hello"},
+			expected: []string{"hello", "Warning, multiple command groups are being remapped to the same command names : \"dummy, dummy\""},
+		},
+		{
+			test: "mapping plugins when conditional on active contexts",
+			pluginVariants: []fakePluginRemapAttributes{
+				fakePluginRemapAttributes{
+					name:    "dummy",
+					target:  configtypes.TargetK8s,
+					aliases: []string{"dum"},
+				},
+				fakePluginRemapAttributes{
+					name:                 "dummy2",
+					target:               configtypes.TargetK8s,
+					invokedAs:            []string{"dummy"},
+					aliases:              []string{"dum"},
+					supportedContextType: []configtypes.ContextType{configtypes.ContextTypeTanzu},
+				},
+			},
+			activeContextType: configtypes.ContextTypeTanzu,
+			args:              []string{},
+			expected:          []string{"dummy2 commands"},
+			unexpected:        []string{"dummy commands"},
+		},
+		{
+			test: "no mapping if active context not one of supportContextType list",
+			pluginVariants: []fakePluginRemapAttributes{
+				fakePluginRemapAttributes{
+					name:    "dummy",
+					target:  configtypes.TargetK8s,
+					aliases: []string{"dum"},
+				},
+				fakePluginRemapAttributes{
+					name:                 "dummy2",
+					target:               configtypes.TargetK8s,
+					invokedAs:            []string{"dummy"},
+					aliases:              []string{"dum"},
+					supportedContextType: []configtypes.ContextType{configtypes.ContextTypeTanzu},
+				},
+			},
+			activeContextType: configtypes.ContextTypeK8s,
+			args:              []string{},
+			expected:          []string{"dummy commands"},
+			unexpected:        []string{"dummy2 commands"},
+		},
+		{
+			test: "nesting plugin within another plugin is not supported",
+			pluginVariants: []fakePluginRemapAttributes{
+				fakePluginRemapAttributes{
+					name:    "dummy",
+					target:  configtypes.TargetK8s,
+					aliases: []string{"dum"},
+				},
+				fakePluginRemapAttributes{
+					name:      "deeper",
+					target:    configtypes.TargetK8s,
+					invokedAs: []string{"kubernetes dummy deeper"},
+				},
+			},
+			args:     []string{"kubernetes", "dummy", "deeper", "say", "hello"},
+			expected: []string{"Remap of plugin into command tree (dummy) associated with another plugin is not supported"},
+		},
+	}
+
+	for _, spec := range tests {
+		env := setupTestCLIEnvironment(t)
+		defer tearDownTestCLIEnvironment(env)
+
+		t.Run(spec.test, func(t *testing.T) {
+			assert := assert.New(t)
+
+			r, w, err := os.Pipe()
+			if err != nil {
+				t.Error(err)
+			}
+			c := make(chan []byte)
+			go readOutput(t, r, c)
+
+			// Set up for our test
+			stdout := os.Stdout
+			stderr := os.Stderr
+			defer func() {
+				os.Stdout = stdout
+				os.Stderr = stderr
+			}()
+			os.Stdout = w
+			os.Stderr = w
+
+			assert.Nil(err)
```

Comment:
Stray assert?

Reply:
Turns out this stray assert was actually a copy/paste from line 755 which has the same "problem" and that git blames me for 😆 

Would you mind also removing the one at line 755 since you have your hands in this file?